code/game/machinery/computer/telecrystalconsoles.dm

```diff
@@ -195,13 +203,10 @@ GLOBAL_LIST_INIT(possible_uplinker_IDs, list("Alfa","Bravo","Charlie","Delta","E
 	if(href_list["scan"])
 		scanUplinkers()
 
-	if(href_list["give1"])
-		var/obj/machinery/computer/telecrystals/uplinker/A = locate(href_list["give1"])
-		A.giveTC(1)
-
-	if(href_list["give5"])
-		var/obj/machinery/computer/telecrystals/uplinker/A = locate(href_list["give5"])
-		A.giveTC(5)
+	if(href_list["give"])
+		var/tcamt = text2num(href_list["give"])
+		var/obj/machinery/computer/telecrystals/uplinker/A = locate(href_list["target"])

```

Comment:
is there a smaller list that contains these that you can locate in ? e.g

    locate(href_list["target"]) in machines

you should also check istype(A) after the locate